### PR TITLE
Allow using a debug Python interpreter

### PR DESCRIPTION
**Pull request by JGamache-autodesk**

### CODE REVIEW

File: source/PyMaterialX/CMakeLists.txt [hunk 1]
```diff
@@ -38,6 +38,14 @@ if(CMAKE_CXX_COMPILER_ID MATCHES Clang)
         add_compile_options(-Wno-undefined-var-template)
     endif()
 endif()
+
+# Postfix rules for Python modules are strict and depend on the presence of a
+# debug interpreter:
+if(PYTHON_IS_DEBUG)
+    set(MATERIALX_PYTHON_DEBUG_POSTFIX "_d")
```
Comment (JGamache-autodesk) [inline source/PyMaterialX/CMakeLists.txt]: Debug interpreter, using python27_d.lib

File: source/PyMaterialX/CMakeLists.txt [hunk 2]
```diff
@@ -38,6 +38,14 @@ if(CMAKE_CXX_COMPILER_ID MATCHES Clang)
...
+else()
+    set(MATERIALX_PYTHON_DEBUG_POSTFIX "")
```
Comment (JGamache-autodesk) [inline source/PyMaterialX/CMakeLists.txt]: Release interpreter using python27.lib

File: source/PyMaterialX/PyBind11/include/pybind11/detail/common.h [hunk 3]
```diff
@@ -103,7 +103,7 @@
 #  endif
 #  pragma warning(push)
 #  pragma warning(disable: 4510 4610 4512 4005)
-#  if defined(_DEBUG)
+#  if defined(_DEBUG) && !defined(Py_DEBUG)
```
Comment (JGamache-autodesk) [inline source/PyMaterialX/PyBind11/include/pybind11/detail/common.h]: Fix was submitted back to PyBind11. See https://github.com/pybind/pybind11/pull/2025

File: source/PyMaterialX/CMakeLists.txt (hunk 1 above)
Comment (ashwinbhat) [reply source/PyMaterialX/CMakeLists.txt]: Is this also the case for Python3 and on other platforms?

Comment (JGamache-autodesk) [reply source/PyMaterialX/CMakeLists.txt]: Confirmed for Python 2.7 version up to 3.1, especially on Windows:
https://github.com/python/cpython/blob/c4cacc8c5eab50db8da3140353596f38a01115ca/PCbuild/readme.txt#L39

Has become a lot more complex following Python 3.2 (or actually a lot simpler by calling `sysconfig.get_config_var('EXT_SUFFIX')`):
Details: https://www.python.org/dev/peps/pep-3149/
Boost::python is aware: https://github.com/boostorg/build/issues/321
Does not appear in PyBind11's issue tracker yet.

And with 3.8 debug and release binaries can coexist peacefully: 
https://docs.python.org/3/whatsnew/3.8.html#debug-build-uses-the-same-abi-as-release-build

Comment (JGamache-autodesk) [reply source/PyMaterialX/CMakeLists.txt]: Thanks @ashwinbhat for asking the right question. I ran tests and PyBind11 implements PEP 3149 for Python 3 perfectly on all test platforms. I made sure the suffix is only used for legacy Python 2.7 when a Debug interpreter is found, on the Windows platform.

**Pull request merged**